packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -458,6 +459,28 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
 
         // Update the used routes in this data store's summarizer node.
         this.summarizerNode.updateUsedRoutes(usedRoutes);
+
+        // If we are loaded, call the channel so it can update the used routes of the child contexts.
+        // If we are not loaded, we will update this when we are realized.
+        if (this.loaded) {
+            this.updateChannelUsedRoutes();
+        }
+    }
+
+    /**
+     * Calls the channel to update used routes of its child contexts.
+     */
+    private updateChannelUsedRoutes() {
+        assert(this.channel !== undefined, "Channel should not be undefined when updating used routes");
+        // Remove the route to this data store, if it exists.
+        const usedChannelRoutes = this.summarizerNode.usedRoutes.filter(
```

Comment:
This function seems a little awkward with the assert and referencing `this.summarizerNode.usedRoutes`, but I understand that in the case of delay-loaded datastores it is necessary without storing it twice.

Maybe at least a comment that `updateChannelUsedRoutes` should be called after `this.summarizerNode.updateUsedRoutes`? or is that already fairly straightforward?

Reply:
The assert is a bit awkward yes. Maybe I can change it to saying that if (!this.loaded) return. Then there is no requirement that this has to be called after the data store is realized.

To answer your other question, there is no requirement that this has to be called after `this.summarizerNode.updateUsedRoutes`. Consider the scenario where we realize a data store during Container load. Here. this method will be called and the used routes will be empty (default value). So, the DDS' used routes will be updated to empty too. But that's okay because GC will be called later and will update the used routes correctly.

Reply:
Note that once we have the check that says GC must have updated used routes before summarize, what I said above will be enforced as well.